ffs-shared/rule/src/commonMain/kotlin/rule/RuleEval.kt

```diff
@@ -113,8 +113,11 @@ private object RuleGrammar : Grammar<RuleExpr<*>>() {
             RuleExpr.NumberExpr(value)
         }
     private val long = optional(minus) and digits map { (minus, int) ->
-        var value = int.text.toLong()
-        minus?.let { value -= value }
+        val value = if (minus != null) {
```

Comment:
This is needed to support the lowest Long value (-9223372036854775808)

Reply:
Oh, nice catch! That `value -= value` was meant to be `value = -value`.